src/src/com/microsoft/aad/adal/AuthenticationActivity.java

```diff
@@ -790,7 +785,7 @@ protected TokenTaskResult doInBackground(String... urlItems) {
                 // Record account in the AccountManager service
                 try {
                     setAccount(result);
-                } catch (Exception exc) {
+                } catch (GeneralSecurityException | IOException exc) {
```

Comment:
So if in the code below:
````
   public void error(String tag, String message, String additionalMessage, ADALError errorCode,
            Throwable err) {
        if (mAndroidLogEnabled) {
            Log.e(tag, getLogMessage(message, additionalMessage, errorCode), err);
        }

        logCommon(tag, message, additionalMessage, LogLevel.Error, errorCode);
    }

```
we append exception information to  `additionalMessage` would it resolve your concerns?